Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,46 @@
-function setAlarm() {}
+let timeRemaining;
+let timer;
+
+document.getElementById("set").addEventListener("click", () => {
+  setAlarm();
+});
+
+document.getElementById("stop").addEventListener("click", () => {
+  clearInterval(timer);
+  document.getElementById("set").innerText = "Set Alarm";
+  document.getElementById("timeRemaining").innerText = "Time Remaining: 00:00";
+  document.getElementById("alarmSet").value = "";
+  audio.pause();
+  audio.currentTime = 0;
+});
+
+function setAlarm() {
+  timeRemaining = parseInt(document.getElementById("alarmSet").value, 10);
+  if (isNaN(timeRemaining) || timeRemaining <= 0) {
+    alert("Please enter a valid positive number");
+    return;
+  }
+  clearInterval(timer); 
+  startTimer(); 
+}
+
+function startTimer() {
```

Comment:
If I put 10 in the box, and press "Set Alarm", I never see the timer set to `00:10` - it looks like it waits for a second, then shows `00:09`. As a user I find this surprising, I'd expect as soon as I click the button it would show the time I put in... Can you make it behave like that?